lib/PhasarLLVM/TypeHierarchy/LLVMTypeHierarchy.cpp

```diff
@@ -96,32 +96,32 @@ std::string
 LLVMTypeHierarchy::removeStructOrClassPrefix(const std::string &TypeName) {
   llvm::StringRef SR(TypeName);
   if (SR.startswith(StructPrefix)) {
-    return SR.ltrim(StructPrefix);
+    return SR.drop_front(StructPrefix.size());
   }
   if (SR.startswith(ClassPrefix)) {
-    return SR.ltrim(ClassPrefix);
+    return SR.drop_front(ClassPrefix.size());
   }
   return TypeName;
 }
 
 std::string LLVMTypeHierarchy::removeTypeInfoPrefix(std::string VarName) {
   llvm::StringRef SR(VarName);
   if (SR.startswith(TypeInfoPrefixDemang)) {
-    return SR.ltrim(TypeInfoPrefixDemang);
```

Comment:
Here `ltrim` is defined as 

```
 LLVM_NODISCARD
     StringRef ltrim(StringRef Chars = " \t\n\v\f\r") const {
       return drop_front(std::min(Length, find_first_not_of(Chars)));
     }
```

Which means that any prefix containing chars in `Chars` will be removed,
If all the chars in SR are included in TypeInfoPrefixDemang, the returned value will be an empty string.
E.g. given, TypeInfoPrefixDemang = "typeinfo for ", and SR = "typeinfo for yointtttttt", the returned value would be empty.

Here what we need is simply remove the the prefix (Chars).

Reply:
A, I see what you mean.